src/adapters/pulseaudio.cpp

```diff
@@ -97,6 +106,87 @@ const string& pulseaudio::get_name() {
   return s_name;
 }
 
+/**
+ * Get device type as string
+ */
+const string pulseaudio::get_device_type() {
+  switch (m_device_type) {
+    case devicetype::SINK:
+      return "sink"s;
+    case devicetype::SOURCE:
+      return "source"s;
+  }
+  throw pulseaudio_error("unreachable");
```

Comment:
Maybe rather have the throw statement in the `default` case and also mention the `m_device_type`.